doc/multi_agent_database_discovery.md

```
@@ -0,0 +1,246 @@
+)
+
+# Agent reads other agents' findings
+findings = catalog_list(kind="statistical_discovery")
+```
+
+## Example Discovery Output
+
+### Database: testdb (E-commerce Order Management)
+
+#### True Statistics (After Deduplication)
+| Metric | Current | Actual |
```

markdownlint-cli2 (0.18.1)

Tables should be surrounded by blank lines

(MD058, blanks-around-tables)